app/models/components/course/achievements_ability_component.rb

```diff
@@ -3,38 +3,34 @@ module Course::AchievementsAbilityComponent
   include AbilityHost::Component
 
   def define_permissions
-    if user
+    if course_user&.student?
```

Comment:
how about move this condition inside `allow_student_show_achievements` and `allow_students_with_achievement_show_badges`.

Otherwise I'd prefer to :
```
if course_user
  # staff will also enter here
  allow_show_achievements
  allow_show_badges
end

if course_user.staff
   allow_manage_achievements
end
```

Reply:
I would prefer the second style, otherwise the condition could be repeated many times in the same file.